### ISSUE TITLE
Analyzer warning when using Unicode Tags.

### ISSUE THREAD

**Issue by daveaglick** [statiqdev/Statiq#997]
### Discussed in https://github.com/orgs/statiqdev/discussions/169

<div type='discussions-op-text'>

<sup>Originally posted by **StevenTCramer** September 17, 2022</sup>
I have added a tag for posts in the Thai language.

When I use `Thai` all is good but when I use `ไทย` then the site works perfectly fine, but the analyzer gives me:

```console
...
 Could not validate relative link: <a href="/tags/&amp;#xE44;&amp;#xE17;&amp;#xE22;" 
...
```

I'm not sure how it is trying to do the analysis of the link.</div>

**Comment by daveaglick** [comment]
Possibly related to #998 which also seems to be dependent on escaping. I'm thinking it's tripping up the way analyzers identify what should be processed and how.

**Comment by daveaglick** [comment]
Verified that the fix for #998 also applies here and added a test to confirm
